WooCommerce/Classes/Model/BetaFeature.swift

```diff
@@ -0,0 +1,115 @@
+import Storage
+
+enum BetaFeature: CaseIterable {
+    case viewAddOns
+    case productSKUScanner
+    case couponManagement
+}
+
+extension BetaFeature {
+    var title: String {
+        switch self {
+        case .viewAddOns:
+            return NSLocalizedString(
+                "View Add-Ons",
+                comment: "Cell title on the beta features screen to enable the order add-ons feature")
+        case .productSKUScanner:
+            return NSLocalizedString(
+                "Product SKU Scanner",
+                comment: "Cell title on beta features screen to enable product SKU input scanner in inventory settings.")
+        case .couponManagement:
+            return NSLocalizedString(
+                "Coupon Management",
+                comment: "Cell title on beta features screen to enable coupon management")
+        }
+    }
+
+    var description: String {
+        switch self {
+        case .viewAddOns:
+            return NSLocalizedString(
+                "Test out viewing Order Add-Ons as we get ready to launch",
+                comment: "Cell description on the beta features screen to enable the order add-ons feature")
+        case .productSKUScanner:
+            return NSLocalizedString(
+                "Test out scanning a barcode for a product SKU in the product inventory settings",
+                comment: "Cell description on beta features screen to enable product SKU input scanner in inventory settings.")
+        case .couponManagement:
+            return NSLocalizedString(
+                "Test out managing coupons as we get ready to launch",
+                comment: "Cell description on beta features screen to enable coupon management")
+        }
+    }
+
+    var settingsKey: WritableKeyPath<GeneralAppSettings, Bool> {
+        switch self {
+        case .viewAddOns:
+            return \.isViewAddOnsSwitchEnabled
+        case .productSKUScanner:
+            return \.isProductSKUInputScannerSwitchEnabled
+        case .couponManagement:
+            return \.isCouponManagementSwitchEnabled
+        }
+    }
+
+    var analyticsStat: WooAnalyticsStat? {
+        switch self {
+        case .viewAddOns:
+            return .settingsBetaFeaturesOrderAddOnsToggled
+        case .productSKUScanner:
+            return nil
+        case .couponManagement:
+            return nil
+        }
+    }
```

Comment:
Done in 6f5edf02eb4497e3d6a6f75ff921179302544432

Reply:
True, although not having a default value means there would be a build error to remind us about adding analytics, which would make it a conscious choice. Since 2 out of 3 are already `nil`, maybe it's not that important, but worth considering.

Reply:
Oh never mind, I just saw what you actually implemented and it makes perfect sense 👌🏽